queries.sql

```diff
@@ -9,3 +9,34 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT deleting_anials;
+UPDATE animals SET weight_kg = weight_kg * - 1;
+ROLLBACK TO deleting_anials;
+UPDATE animals SET weight_kg = weight_kg * - 1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, AVG(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MAX(weight_kg), MIN(weight_kg) FROM animals GROUP BY species;
```

Comment:
- You have done great so far :+1: :+1: But there are a few queries you missed. :+1: :+1: 
> What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- [ ] You didn't write the query for this requirement above, I'll suggest you complete this. :+1: :100: